parquet-hadoop/src/test/java/org/apache/parquet/hadoop/TestDataPageChecksums.java

```diff
@@ -688,8 +690,13 @@ private int getDataOffset(Page page) {
    */
   private ParquetFileReader getParquetFileReader(Path path, Configuration conf, List<ColumnDescriptor> columns)
       throws IOException {
-    ParquetMetadata footer = ParquetFileReader.readFooter(conf, path);
-    return new ParquetFileReader(conf, footer.getFileMetaData(), path, footer.getBlocks(), columns);
+    HadoopInputFile inputFile = HadoopInputFile.fromPath(path, conf);
+    SeekableInputStream inputStream = inputFile.newStream();
+    ParquetReadOptions readOptions = HadoopReadOptions.builder(conf).build();
+    ParquetMetadata footer = ParquetFileReader.readFooter(inputFile, readOptions, inputStream);
+    ParquetFileReader reader = ParquetFileReader.open(inputFile, footer, readOptions, inputStream);
+    reader.setRequestedSchema(columns);
+    return reader;
```

Comment:
It would be better to explicitly add a new test case rather than silently changing old ones.

Reply:
I understand that, but it seems there is no dedicated test suite for `ParquetFileReader` constructor. The replaced API is deprecated, and the replacement does not reduce the test coverage because the deprecated API is used many times in other test suites.